xui/additional_files_unzipped/s3_browser/s3_browser/aws_bucket_browser_operations/aws_bucket_browser_operations.py

```diff
@@ -261,15 +261,15 @@ def is_versioning_enabled(s3_client, resource_id, default_region):
         return False
 
     except (CustomField.DoesNotExist, CustomFieldValue.DoesNotExist) as error:
-        # This error can arise when s3 buckets are added from discovery. In which case, attempt to find versioning a last with using the default region
-        logger.error(f"CustomField 's3_bucket_region' does not exist. Trying default_region={default_region}")
+        # This error can arise when s3 buckets are added from discovery and their s3_bucket_region parmeter is not set. As a backup, attempt to find versioning info using the default region
+        logger.warning(f"CustomField 's3_bucket_region' does not exist. Trying default_region={default_region}")
```

Comment:
It would be good to mention that this is for fetching versioning info, so they know what to expect to work and what not to work if their buckets are not in the default region.

Reply:
I have this in there already 
> "As a backup, attempt to find versioning info using the default region" 
but I will try and make this clearer and more accurate

Reply:
![Screenshot 2023-12-18 at 2 26 57 PM](https://github.com/CloudBoltSoftware/applet-s3-bucket-browser/assets/105660893/9d0ac94d-1ef1-400f-8396-f1e96813e9ac)

Reply:
You have that in a comment, but users cannot see that when they are looking at the log file. They may be able to trace the issue back to the IWH in question and find that comment, but most will not succeed in that. Any info that would be helpful to a CB Admin in t-shooting should be moved from comment --> log message.

Reply:
WHOOPS sorry I also added this at the start of the function
![Screenshot 2023-12-18 at 2 35 10 PM](https://github.com/CloudBoltSoftware/applet-s3-bucket-browser/assets/105660893/e873262a-28c9-42b2-8d8e-8006e7e76e2b)

Reply:
ah I can make those log messages as well

Reply:
OK new version
![Screenshot 2023-12-18 at 2 38 28 PM](https://github.com/CloudBoltSoftware/applet-s3-bucket-browser/assets/105660893/34766583-40ef-4168-9328-d41ad172bb79)

Reply:
![Screenshot 2023-12-18 at 2 51 33 PM](https://github.com/CloudBoltSoftware/applet-s3-bucket-browser/assets/105660893/63108fb3-c1e9-4139-8a46-ab39f939d1cf)


ok log when there's no `s3_bucket-region` but default works
```
2023-12-18 19:49:48,712 [INFO] shared_modules.aws_bucket_browser_operations: Checking if Bucket Versioning is enabled for the bucket 'ui-demo-experience' 
2023-12-18 19:49:48,718 [WARNING] shared_modules.aws_bucket_browser_operations: CustomField 's3_bucket_region' does not exist
2023-12-18 19:49:48,719 [INFO] shared_modules.aws_bucket_browser_operations: This error can arise when s3 buckets are added from discovery and their s3_bucket_region parmeter is not set.
2023-12-18 19:49:48,719 [INFO] shared_modules.aws_bucket_browser_operations: As a backup, attempting to connect to the AWS Resource client using the default region='us-east-1'.
2023-12-18 19:49:50,127 [INFO] shared_modules.aws_bucket_browser_operations: Connected to the AWS Resource client
```

And new log when the default also does not work (aka botocore error)
```
2023-12-18 19:47:49,178 [INFO] shared_modules.aws_bucket_browser_operations: Checking if Bucket Versioning is enabled for the bucket 'ui-demo-experience' 
2023-12-18 19:47:49,186 [WARNING] shared_modules.aws_bucket_browser_operations: CustomField 's3_bucket_region' does not exist
2023-12-18 19:47:49,187 [INFO] shared_modules.aws_bucket_browser_operations: This error can arise when s3 buckets are added from discovery and their s3_bucket_region parmeter is not set.
2023-12-18 19:47:49,191 [INFO] shared_modules.aws_bucket_browser_operations: As a backup, attempting to connect to the AWS Resource client using the default region='ap-east-1'.
2023-12-18 19:47:51,052 [INFO] common.methods: Exception in plug-in S3 Browser, line 111:
  "versioning_enabled": operations.is_versioning_enabled(s3_client,resource_id,default_region=bucket_location),
ClientError: An error occurred (IllegalLocationConstraintException) when calling the GetBucketVersioning operation: The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.
2023-12-18 19:47:51,054 [ERROR] api.v3.errors: Unhandled exception in the CB API code:
Traceback (most recent call last):
  File "/var/www/html/cloudbolt/static/uploads/shared_modules/aws_bucket_browser_operations.py", line 254, in is_versioning_enabled
    region_attribute = resource.attributes.get(field_id=s3rcf.id)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/ec2-user/Git/cloudbolt/src/orders/models.py", line 245, in get
    return super(CustomFieldValueQuerySet, self).get(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/db/models/query.py", line 435, in get
    raise self.model.DoesNotExist(
orders.models.CustomFieldValue.DoesNotExist: CustomFieldValue matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/ec2-user/Git/cloudbolt/src/cbhooks/api/v3/viewsets/inbound_web_hook.py", line 93, in run
    ret = iwh.run_hook(method=request.method, parameters=parameters, files=files)
  File "/home/ec2-user/Git/cloudbolt/src/cbhooks/models.py", line 2521, in run_hook
    return self.hook.cast().run_without_retries(
  File "/home/ec2-user/Git/cloudbolt/src/cbhooks/models.py", line 3683, in run_without_retries
    return_value = callable_func(*args, **kwargs)
  File "<string>", line 111, in inbound_web_hook_post
  File "/var/www/html/cloudbolt/static/uploads/shared_modules/aws_bucket_browser_operations.py", line 272, in is_versioning_enabled
    if s3_bucket.BucketVersioning(resource.name).status == "Enabled":
  File "/usr/local/lib/python3.9/site-packages/boto3/resources/factory.py", line 339, in property_loader
    self.load()
  File "/usr/local/lib/python3.9/site-packages/boto3/resources/factory.py", line 505, in do_action
    response = action(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(*args, **params)
  File "/usr/local/lib/python3.9/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python3.9/site-packages/botocore/client.py", line 676, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (IllegalLocationConstraintException) when calling the GetBucketVersioning operation: The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.
```

Reply:
The success & error cases look good there!